File: introduction/HJ101/HJ101.cpp
#include <cstdio>
#include <iostream>
#include<vector>
#include<algorithm>
using namespace std;

int main() {
    int a, b;
    while (cin >> a ) { 
        vector<int>arr(a);
        for(int i=0;i<a;i++){
            scanf("%d",&arr[i]);
        }
        int flag;
        scanf("%d",&flag);
        auto f= [flag](int a,int b)  { 
            if(flag==1) 
                return a>b;
            return a<=b;
        };  
        sort(arr.begin(),arr.end(),f);
        for(int i=0;i<arr.size();i++){
            printf("%d ",arr[i]);
        }
    }
    return 0;
}

File: introduction/HJ101/HJ101.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var a int
	for {
		_, err := fmt.Scan(&a)
		if err != nil {
			break
		}

		arr := make([]int, a)
		for i := 0; i < a; i++ {
			fmt.Scan(&arr[i])
		}

		var flag int
		fmt.Scan(&flag)

		sort.Slice(arr, func(i, j int) bool {
            if flag == 0 {
                return arr[i]<=arr[j]
            }
			return arr[i]>arr[j]
		})

		for i := 0; i < len(arr); i++ {
			fmt.Printf("%d ", arr[i])
		}
	}
}
